README.md

```diff
@@ -34,17 +57,57 @@ go get github.com/libp2p/go-libp2p-pubsub
 
 To be used for messaging in p2p instrastructure (as part of libp2p) such as IPFS, Ethereum, other blockchains, etc.
 
-## Implementations
+### Examples
 
-See [libp2p/specs/pubsub#Implementations](https://github.com/libp2p/specs/tree/master/pubsub#Implementations).
+`To be written`
 
 ## Documentation
 
-See the [libp2p specs](https://github.com/libp2p/specs/tree/master/pubsub) for high level documentation
-and [godoc](https://godoc.org/github.com/libp2p/go-libp2p-pubsub) for API documentation.
+See the [libp2p specs](https://github.com/libp2p/specs/tree/master/pubsub) for high level documentation and [godoc](https://godoc.org/github.com/libp2p/go-libp2p-pubsub) for API documentation.
 
+### In this repo, you will find
 
-## Tracing
+See [libp2p/specs/pubsub#Implementations](https://github.com/libp2p/specs/tree/master/pubsub#Implementations).
+
+```
+.
+├── LICENSE
+├── README.md
+# Regular Golang repo set up
+├── codecov.yml
+├── pb
+├── go.mod
+├── go.sum
+├── doc.go
+# Floodsub specific
+├── floodsub.go
+├── floodsub_test.go
+# Randomsub specific
+├── randomsub.go
+# Gossipsub specific
+├── blacklist.go
+├── blacklist_test.go
+├── comm.go
+├── discovery.go
+├── discovery_test.go
+├── gossipsub.go
+├── gossipsub_test.go
+├── mcache.go
+├── mcache_test.go
+├── notify.go
+# Used by all
+├── pubsub.go
+├── sign.go
+├── sign_test.go
+├── subscription.go
+├── topic.go
+├── topic_test.go
+├── trace.go
+├── tracer.go
+└── validation.go
+```
```

Comment:
@vyzo did I get this separation right?

Reply:
PS: I would like to break them into separate folders post https://github.com/libp2p/go-libp2p-pubsub/pull/263 is merged

Reply:
no, blacklist, comm, discovery, and notify are used by all, they are not gossipsub specific.
The only thing that is gossipsub specific is `gossipsub.go` and `mcache.go`.

Reply:
Also, we can't break into separate folders without major refactoring, the routers need access to pubsub protected members.

Reply:
plus you shouldn't put the test files in there, they are irrelevant for presentation purposes.

Reply:
Thank you! Fair points! Updated